compiler/test/Dialect/Linalg/fuse-attention.mlir

```diff
@@ -160,3 +160,82 @@ transform.sequence failures(propagate) {
   %0 = transform.structured.match attributes {__root__} in %arg0 : (!pdl.operation) -> !pdl.operation
   %transformed, %loops:2 = transform.structured.fuse_ext %0 {tile_interchange = [0, 1, 4, 3, 2], tile_sizes = [2, 4, 0, 0, 0]}
 }
+
+// -----
+
+#map = affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d2, d3)>
+#map1 = affine_map<(d0, d1, d2, d3, d4) -> (d0, d3, d4)>
+#map2 = affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d2, d4)>
+
+// batch size = 4
+// sequence length = 1024
+// hidden size = 512
+// head number = 8
+// head dimension = 64
+func.func @multiquery_attention_with_prologue_proj(%arg0: tensor<4x1024x512xf32>, %arg1: tensor<512x512xf32>, %arg2: tensor<512x64xf32>, %arg3: tensor<512x64xf32>) -> tensor<4x8x1024x64xf32> {
```

Comment:
Just a suggestion. Can we add math form of this attention?
Such as ones in https://arxiv.org/pdf/1911.02150.pdf

Reply:
Adding math forms is a bit difficult in text format. I only added the link here :)